### ISSUE TITLE
Possible for photogate timer to run to infinity.

### ISSUE THREAD

**Issue by bryo5363** [phetsims/pendulum-lab#2]
Steps to reproduce
-Maximize length of string.
-Before dropping, make sure you have the photogate timer accessed and press start. 
-Drop from around 115º
-When the mass hits the point of minimum PE, minimize the length of the string.
-The mass will start to circumvent around the holding point. The timer will also continue to run. 

![screen shot 2014-09-23 at 12 29 40 pm](https://cloud.githubusercontent.com/assets/7717368/4377574/287e06d4-4350-11e4-9e9c-b9fc32cdba58.png)

When you combine this issue with the last issue, you start to observe some weird behavior.

![screen shot 2014-09-23 at 12 34 23 pm](https://cloud.githubusercontent.com/assets/7717368/4377584/4981ea9e-4350-11e4-91c6-aa03e2069437.png)

**Comment by arouinfar** [comment]
While playing around with dev.3 running OS X 10.9.5 and Chrome 40.0.2214.91, I wasn't able to reproduce what @bryo5363 has reported. However, I have found a few ways to make the period timer run to infinity.

<b>Method 1:</b>
1) While pendulum 1 is swinging, start the period timer (period timer should be set to record pendulum 1)
2) Before the period has been fully recorded, switch the timer to pendulum 2
(This method works no matter what state pendulum 2 is in. It can be turned off, swinging, or stationary.)

<b>Desired behavior:</b>
If Pendulum 2 is off-screen or stationary, the timer should reset itself to show 0.0000 s and the period trace should immediately disappear (instead of fading away).
If Pendulum 2 is in motion, the period trace for the 1st pendulum should immediately disappear and the timer should start recording/tracing pendulum 2's period.

<b>Method 2:</b>
1) While a pendulum is swinging, start the period timer (period timer should be set to appropriate pendulum)
2) Click on the return button. The pendulum stops swinging, but the timer continues to record.

<b>Desired behavior:</b>
When the return button is pressed, the period trace should immediately disappear (instead of fading as usual) and the timer should display 0.0000 s.

-----
Troubleshooting information (do not edit):
Name: Pendulum Lab
URL: http://www.colorado.edu/physics/phet/dev/html/pendulum-lab/1.0.0-dev.3/pendulum-lab_en.html
Version: 1.0.0-dev.3
Features missing: touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.91 Safari/537.36
Language: en-us
Window: 1073x700
Pixel Ratio: 2/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 31 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 16)
Max viewport: 16384x16384
OES_texture_float: true

**Comment by andrey-zelenkov** [comment]
Significant changes were made for period timer. Fixed return button behavior (it's stop and display 0.0000 s after pressing 'return' button). Prevented switching to the second pendulum if it's not visible (I've tried simply to switch back to the first pendulum but error occurred in SCENERY/RelativeTransform.js). If second pendulum is visible and we switch to it then period trace for the 1st pendulum should immediately disappear and the timer start recording/tracing pendulum 2's period. Reassign to @arouinfar for verification.
